### ISSUE TITLE
the itk 5.3rc4post2 package on pip cannot be installed if an earlier (5.2) version is already installed

### ISSUE THREAD

**Issue by HastingsGreer** [InsightSoftwareConsortium/ITK#3592]
### Description

When pip is used to install 5.3rc4.post2, if itk 5.2 is previously installed, then the resulting installation is non-functional.

### Steps to Reproduce

```
pip install itk
pip install itk==5.3rc4.post2
```
```
import itk
itk.MedianImageFilter
```

Here is a colab notebook demonstrating the issue

[notebook](https://colab.research.google.com/drive/13Gj0v8BgNUgVdmIZuHoYiSw5eP4I2cn1?usp=sharing)

### Expected behavior

I expect running pip install itk==5.3rc4.post2 to uninstall or overwrite the previous version of itk

### Actual behavior

the line `itk.MedianImageFilter`
throws the error message 
```
KeyError                                  Traceback (most recent call last)

[<ipython-input-4-da83592e4888>](https://localhost:8080/#) in <module>
----> 1 itk.MedianImageFilter
[/usr/local/lib/python3.7/dist-packages/itk/support/lazy.py](https://localhost:8080/#) in __getattribute__(self, attr)
    136                     module = self.__belong_lazy_attributes[attr]
    137                     namespace = {}
--> 138                     base.itk_load_swig_module(module, namespace)
    139                     self.loaded_lazy_modules.add(module)
    140                     for k, v in namespace.items():

[/usr/local/lib/python3.7/dist-packages/itk/support/base.py](https://localhost:8080/#) in itk_load_swig_module(name, namespace)
     94         deps = l_data.get_module_dependencies()
     95         for dep in deps:
---> 96             itk_load_swig_module(dep, namespace)
     97 
     98     # It is possible that template_feature instantiations from this module

[/usr/local/lib/python3.7/dist-packages/itk/support/base.py](https://localhost:8080/#) in itk_load_swig_module(name, namespace)
     90     # via the itk_base_global_module_data instance defined at the bottom of this file, which
     91     # knows how to find those configuration files.
---> 92     l_data = itk_base_global_module_data[name]
     93     if l_data:
     94         deps = l_data.get_module_dependencies()

KeyError: 'ITKImageSources'
```

### Reproducibility
Reproduces every time

### Versions
The first version with this issue is itk 5.3rc4. For previous versions up to 5.3rc3, installing the new version successfully overwrites the old version.

### Environment
I have verified this issue on ubuntu, windows, and google colab

**Comment by HastingsGreer** [comment]
@thewtex Right now oai_analysis_2 is pointing to an old version of icon_registration, upgrading to the new version of icon_registration is blocked by this issue. The alternative is to go through and be careful with install orders to make sure that itk 5.2 is never installed before icon_registration, but that will be fragile

**Comment by HastingsGreer** [comment]
Duplicate of #3175
